src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -199,12 +199,25 @@ def execute(self, args):
                                   EVENT_INVOKER_FILTER_RESULT)
         from knack.util import CommandResultItem, todict
         from azure.cli.core.commands.events import EVENT_INVOKER_PRE_CMD_TBL_TRUNCATE
+        from azure.cli.core.commands import cache
 
         # TODO: Can't simply be invoked as an event because args are transformed
         args = _pre_command_table_create(self.cli_ctx, args)
 
         self.cli_ctx.raise_event(EVENT_INVOKER_PRE_CMD_TBL_CREATE, args=args)
-        self.commands_loader.load_command_table(args)
+        success = False
+        if cache.cache_exists():
+            logger.debug("attempting to use cache")
+            success = cache.load_command_table(self.commands_loader, args) is not None
```

Comment:
I'd recommend a simply try/except here where the method will throw if the cache does not exist.